lib/data/wrapper.js

```diff
@@ -96,31 +96,35 @@ const data = {
         // If objectGetInfo.key exists the md-model-version is 2 or greater.
         // Otherwise, the objectGetInfo is just the key string.
         const key = objectGetInfo.key ? objectGetInfo.key : objectGetInfo;
-        log.debug('sending delete to datastore', { implName, key,
+        log.debug('sending delete to dataStore', { implName, key,
             method: 'delete' });
         client.delete(key, log.getSerializedUids(), err => {
             if (err) {
-                log.error('error from sproxyd', { error: err });
+                log.error('error deleting object from dataStore',
+                    { error: err }, { location: key });
                 return cb(errors.InternalError);
             }
+            log.trace('delete object successful');
             return cb();
         });
     },
 
     // It would be preferable to have an sproxyd batch delete route to
     // replace this
-    batchDelete: (locations, log) => {
+    batchDelete: (locations, log, cb) => {
         async.eachLimit(locations, 5, (loc, next) => {
-            data.delete(loc, log, err => {
-                if (err) {
-                    log.error('one part of a batch delete failed',
-                    { location: loc });
-                }
-                return next();
-            });
+            data.delete(loc, log, next);
         },
-        () => {
-            log.end();
+        err => {
+            const callback = cb || function _noop() {
+                return log.end();
+            };
```

Comment:
Hmm... our linting supreme leader will not like this: making unnecessary new functions. It should better be `const callback = cb || log.end` if `log.end` doesn't take any argument or you have to use ternary in the returns below.